### ISSUE TITLE
Button Toggle Group

### ISSUE THREAD

**Issue by feloy** [containers/podman-desktop#4985]
### Is your feature request related to a problem? Please describe

Related to #4707

To implement the toggle buttons to select the severity to display, we need to have a "Button Toggle Group" component (similar to https://material.angular.io/components/button-toggle/overview).

![image-checker-design](https://github.com/containers/podman-desktop/assets/9973512/3fd15cbf-257e-4f8b-a4ce-70afb48d57ba)

### Describe the solution you'd like

The interface for the component would be:

```
const buttons: ToggleButton[] = [
{
  icon: faExclamationCircle,
  iconClass: 'text-red-600',
  name: 'Critical (5)',
  checked: true,
  disabled: false
}, {
  icon: faExclamationTriangle,
  iconClass: 'text-amber-500',
  name: 'High (2)',
  checked: true,
  disabled: false
}, {
  icon: faExclamationTriangle,
  iconClass: 'text-gray-800',
  name: 'Medium (0)',
  checked: false,
  disabled: true
}, ...
];

function onChange(newState: ToggleButton[]) {
 // newState contains updated checked values for buttons
}

<ToggleButtonGroup buttons={buttons} on:change={(event) => onChange(event.details)} />
```

**Comment by benoitf** [comment]
should we have the declarative button definition or just something more in line with svelte components

```svelte
<ToggleButtonGroup>
  <ToggleButton icon={faExclamationCircle} class="text-red-600">Critical ${data.length}</ToggleButton>
  <ToggleButton icon={...} class="text-red-600">Critical ${data.length}</ToggleButton>
</ToggleButtonGroup>
```svelte

**Comment by feloy** [comment]
Yes, it seems a more elegant way.

But I don't have a clue how to implement it this way. For example, how the group component can know about the children buttons, which one is the first one, and which one the last one, to draw them differently (corners being rounded only at beginning and end of the group)?

**Comment by benoitf** [comment]
maybe it's not possible :-)

if it might help:
in tailwindcss you have `first` and `last` directive to know if you're the first or last child
https://tailwindcss.com/docs/hover-focus-and-other-states#first-last-odd-and-even

you can bind children using `bind:this={yourArray[i]} `

but I'm not sure about why, etc

you can also displatch stuff 

in your case you can use setContext/getContext https://svelte.dev/docs/svelte#setcontext between component and children, I mean I have not looked deeply of what you need but between svelte and tailwind there are some capabilities.

**Comment by feloy** [comment]
I will first explore the direction you proposed, to check if it is possible.

I have found this solution using contexts (https://github.com/sveltejs/svelte/issues/5381#issuecomment-690842511), which could help.

Thanks

Where would you place the event handler, in the group or in each button?

```
<ToggleButtonGroup on:change={handleAllCheckedButtons}>
  <ToggleButton on:checked={handleButton1Checked} icon={faExclamationCircle} class="text-red-600">Critical ${data.length}</ToggleButton>
  <ToggleButton on:checked={handleButton2Checked} icon={...} class="text-red-600">Critical ${data.length}</ToggleButton>
</ToggleButtonGroup>
```

For this particular case, I have a preference for placing it into the group, because all the buttons are very related, and this would save having very similar handlers. But I'm not sure if it is a common case.

**Comment by feloy** [comment]
Fixed by #5000
